pydantic_ai_slim/pydantic_ai/_utils.py

```diff
@@ -71,6 +71,56 @@ def is_model_like(type_: Any) -> bool:
     )
 
 
+def resolve_json_schema_refs(schema: JsonSchemaValue) -> JsonSchemaValue:
+    """Recursively resolve all $ref references in a JSON schema by inlining them.
+    
+    Args:
+        schema: The JSON schema to resolve references in
+        
+    Returns:
+        A new JSON schema with all $ref references resolved and inlined
+    """
+    if not isinstance(schema, dict):
+        return schema
+    
+    # If there are no $refs in the schema, return as-is
+    if '$defs' not in schema and '$ref' not in str(schema):
+        return schema
+    
+    # Get the definitions from the schema
+    defs = schema.get('$defs', {})
+    
+    def resolve_refs(obj: Any) -> Any:
```

Comment:
AI slob below, I really think this shouldn't be necessary but then I also don't really know what the StructuredDict is for or why it's needed. 


FYI we ran into this because StructuredDict is not pickl'able (makes joblib memory caching impossible) and we tried finding a work around through a plain jsonschema.